csharp/src/Apache.Arrow/Arrays/StringArray.cs

```diff
@@ -164,5 +164,30 @@ IEnumerator<string> IEnumerable<string>.GetEnumerator()
         }
 
         IEnumerator IEnumerable.GetEnumerator() => ((IEnumerable<string>)this).GetEnumerator();
+
+        int ICollection<string>.Count => Length;
+        bool ICollection<string>.IsReadOnly => true;
+        void ICollection<string>.Add(string item) => throw new NotSupportedException("Collection is read-only.");
+        bool ICollection<string>.Remove(string item) => throw new NotSupportedException("Collection is read-only.");
+        void ICollection<string>.Clear() => throw new NotSupportedException("Collection is read-only.");
+
+        bool ICollection<string>.Contains(string item)
```

Comment:
This could be optimized by using a similar null vs not-null fork and by using the materializedStringStore (if present) or converting the search string to binary and doing a binary comparison if the string list hasn't been materialized.